src/Interceptor/Interceptor/CacheInterceptor.php

```diff
@@ -109,36 +131,69 @@ public function getSuffixPriority(): int
         return 20;
     }
 
-    private function buildCacheKey(Instance $instance, Cache $annotation): string
+    private function buildCacheKey(Instance $instance, Cache $attribute): string
     {
-        $version = $annotation->getVersion() !== null ? '.v' . $annotation->getVersion() : null;
+        $method = $instance->getMethod();
 
-        if ($annotation->getId() !== null) {
-            $parameters = $instance->getMethod()
-                ->getParameters();
+        $parameters = $method->getParameters();
 
-            return $this->resolveExpression($annotation->getId(), $parameters) . $version;
+        $identifier = $instance->getReflection()
+            ->getName() . '.' . $method->getName();
+
+        if (\count($parameters) > 0) {
+            foreach ($parameters as $parameterName => $parameterValue) {
+                $identifier .= '.' . $parameterName . '.' . hash('xxh3', serialize($parameterValue));
+            }
         }
 
-        return $this->buildDefaultIdentifier($instance, $annotation) . $version;
+        $identifier = array_reduce(
+            $this->getTags($instance, $attribute),
+            static fn (string $identifier, string $tag) => $identifier .= '.' . $tag,
+            $identifier
+        );
+
+        $identifier .= $this->getMethodCode($method->getReflection());
+
+        return hash('xxh3', $identifier);
     }
 
-    private function buildDefaultIdentifier(Instance $instance, Cache $annotation): string
+    private function getMethodCode(\ReflectionMethod $reflectionMethod): string
```

Comment:
@arnaud-23 Regarding performance of the `file` method, I did a quick benchmark with both average (around 100 lines) and very long file (around 4k):

```php
foreach ([$smallFile, $longFile] as $file) {
    for ($i = 0; $i <= 10 ; $i++) {
        $start = microtime(true);
        file($file);
        $end = microtime(true);
        $executionTime = ($end - $start) * 1000;

        echo "Execution time: $executionTime";
    }
}
```
Results (in milliseconds):
```
// Small file
Execution time: 0.026941299438477
Execution time: 0.014066696166992
Execution time: 0.012874603271484
Execution time: 0.012874603271484
Execution time: 0.011920928955078
Execution time: 0.015974044799805
Execution time: 0.012874603271484
Execution time: 0.013828277587891
Execution time: 0.015020370483398
Execution time: 0.015974044799805
Execution time: 0.016212463378906

// Long file
Execution time: 0.14305114746094
Execution time: 0.13303756713867
Execution time: 0.12898445129395
Execution time: 0.12397766113281
Execution time: 0.12803077697754
Execution time: 0.12612342834473
Execution time: 0.12016296386719
Execution time: 0.12493133544922
Execution time: 0.12302398681641
Execution time: 0.12302398681641
Execution time: 0.12302398681641
```

I also did the same benchmark with the whole proxy stack
```php
$proxy1 = $proxyFactory->createProxy(new ClassWithCacheAttributes());
$proxy2 = $proxyFactory->createProxy(new VeryLongFile());

foreach ([$proxy1, $proxy2] as $proxy) {
    for ($i = 0; $i <= 10 ; $i++) {
        $start = microtime(true);
        $proxy->methodWithAttribute();
        $end = microtime(true);
        $executionTime = ($end - $start) * 1000;

        echo "Execution time: $executionTime\n";
    }
}
```
Results (in milliseconds):
```
// Small file 
Execution time: 0.26416778564453 // first execution is slower since it initialises the cache
Execution time: 0.050067901611328
Execution time: 0.054121017456055
Execution time: 0.037908554077148
Execution time: 0.037908554077148
Execution time: 0.036954879760742
Execution time: 0.036954879760742
Execution time: 0.041007995605469
Execution time: 0.037908554077148
Execution time: 0.041961669921875
Execution time: 0.040054321289062

// Long file
Execution time: 0.37908554077148 // idem
Execution time: 0.16212463378906
Execution time: 0.15521049499512
Execution time: 0.15091896057129
Execution time: 0.14686584472656
Execution time: 0.14686584472656
Execution time: 0.14281272888184
Execution time: 0.14090538024902
Execution time: 0.14591217041016
Execution time: 0.14495849609375
Execution time: 0.14591217041016
```